src/js/components/Form/ProjectPicker.jsx

```diff
@@ -0,0 +1,167 @@
+import PropTypes from 'prop-types'
+import React, { useContext, useEffect, useState } from 'react'
+
+import { Context } from '../../state'
+import { httpGet, httpRequest, requestOptions } from '../../utils'
+import { useTranslation } from 'react-i18next'
+
+function ProjectPicker({
+  disabled,
+  name,
+  onChange,
+  onError,
+  readOnly,
+  required,
+  value
+}) {
+  const [globalState, dispatch] = useContext(Context)
+  const [namespaces, setNamespaces] = useState([])
+  const [projectTypes, setProjectTypes] = useState([])
+  const [projects, setProjects] = useState([])
+  const [namespaceID, setNamespaceID] = useState()
+  const [projectTypeID, setProjectTypeID] = useState()
+
+  const { t } = useTranslation()
+
+  useEffect(() => {
+    const namespacesURL = new URL('/namespaces', globalState.baseURL)
+    const projectTypesURL = new URL('/project-types', globalState.baseURL)
+
+    Promise.all([
+      httpRequest(globalState.fetch, namespacesURL, requestOptions),
+      httpRequest(globalState.fetch, projectTypesURL, requestOptions)
+    ]).then(([namespacesResponse, projectTypesResponse]) => {
+      if (!namespacesResponse.success) {
+        onError(namespacesResponse.data)
+      } else if (!projectTypesResponse.success) {
+        onError(projectTypesResponse.data)
+      } else {
+        setNamespaces(namespacesResponse.data)
+        setProjectTypes(projectTypesResponse.data)
+        if (value) {
+          setNamespaceID(value.namespace_id)
+          setProjectTypeID(value.project_type_id)
+        }
+      }
+    })
+  }, [])
+
+  useEffect(() => {
+    if (!projectTypeID || !namespaceID) {
+      setProjects([])
+      return
+    }
+
+    const projectsURL = new URL(
+      `/projects?namespace_id=${encodeURIComponent(
+        namespaceID
+      )}&project_type_id=${encodeURIComponent(projectTypeID)}`,
+      globalState.baseURL
+    )
```

Comment:
I was considering changing our HTTP wrapper to do something useful with search parameters.  I would use a `URLSearchParams` instance instead of encoding them.
```javascript
const projectURL = new URL("/projects", globalState.baseURL)
const params = new URLSearchParams({namespace_id:namespaceID, project_type_id:projectTypeID})
projectURL.search = params.toString()
```

I think that the `toString()` call is necessary though most browsers are fine with assigning a `URLSearchParams` instance to `url.query`.  AFAICT, the [URL Standard](https://url.spec.whatwg.org/#ref-for-dom-url-search①) requires it to be a string by YMMV